src/System.IO/tests/BinaryWriter/BinaryWriter_WriteByteCharTests.cs

```diff
@@ -0,0 +1,571 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Xunit;
+using System;
+using System.IO;
+using System.Text;
+
+namespace BinaryWriterTests
+{
+    public class BinaryWriter_WriteByteCharTests
+    {
+        /// <summary>
+        /// Cases Tested:
+        /// 1) Tests that BinaryWriter properly writes chars into a stream.
+        /// 2) Tests that if someone writes surrogate characters, an argument exception is thrown
+        /// 3) Casting an int to char and writing it, works.
+        /// </summary>
+        [Fact]
+        public static void BinaryWriter_WriteCharTest()
+        {
+            MemoryStream mstr = new MemoryStream();
+            BinaryWriter dw2 = new BinaryWriter(mstr);
+            BinaryReader dr2 = new BinaryReader(mstr);
+
+            Char[] chArr = new Char[0];
+            int ii = 0;
+
+            // [] Write a series of characters to a MemoryStream and read them back
+            chArr = new Char[] { 'A', 'c', '\0', '\u2701', '$', '.', '1', 'l', '\u00FF', '\n', '\t', '\v' };
+            for (ii = 0; ii < chArr.Length; ii++)
+                dw2.Write(chArr[ii]);
+
+            dw2.Flush();
+            mstr.Position = 0;
+            try
+            {
+                for (ii = 0; ; ii++)
+                {
+                    char c = dr2.ReadChar();
+                    Assert.Equal(chArr[ii], c);
+                }
+            }
+            catch (EndOfStreamException)
```

Comment:
If we are expecting this exception every time, we should Assert for it instead of just catching it